core/domain/story_domain.py

```diff
@@ -0,0 +1,369 @@
+            story_last_updated: datetime.datetime. Date and time when
+                the story was last updated.
+
+        Returns:
+            Story. The corresponding Story domain object.
+        """
+        story = cls(
+            story_dict['id'], story_dict['title'],
+            story_dict['description'], story_dict['notes'],
+            [
+                StoryNode.from_dict(node_dict)
+                for node_dict in story_dict['story_nodes']
+            ], story_dict['schema_version'], story_dict['language_code'],
+            story_dict['version'], story_created_on, story_last_updated)
+
+        return story
+
+    @classmethod
+    def create_default_story(
+            cls, story_id, title=feconf.DEFAULT_STORY_TITLE,
+            description=feconf.DEFAULT_STORY_DESCRIPTION,
+            notes=feconf.DEFAULT_STORY_NOTES,
+            language_code=constants.DEFAULT_LANGUAGE_CODE):
+        """Returns a story domain object with default values.
+
+        Args:
+            story_id: str. The unique id of the story.
+            title: str. The title of the story.
+            category: str. The category of the story.
+            objective: str. The objective of the story.
+            language_code: str. The language code of the story (like 'en'
+                for English).
+
+        Returns:
+            Story. The Story domain object with the default values.
+        """
+        return cls(
+            story_id, title, description, notes, [],
+            feconf.CURRENT_STORY_SCHEMA_VERSION, language_code, 0)
+
+    @classmethod
+    def update_story_contents_from_model(
+            cls, versioned_story_contents, current_version):
+        """Converts the states blob contained in the given
+        versioned_story_contents dict from current_version to
+        current_version + 1. Note that the versioned_story_contents being
+        passed in is modified in-place.
+
+        Args:
+            versioned_story_contents: dict. A dict with two keys:
+                - schema_version: str. The schema version for the
+                    story_contents dict.
+                - story_contents: dict. The dict comprising the story
+                    contents.
+            current_version: int. The current schema version of story_contents.
+
+        Raises:
+            Exception: The value of the key 'schema_version' in
+            versioned_story_contents is not valid.
+        """
+        if (versioned_story_contents['schema_version'] + 1 >
+                feconf.CURRENT_STORY_SCHEMA_VERSION):
+            raise Exception('story is version %d but current story'
+                            ' schema version is %d' % (
+                                versioned_story_contents['schema_version'],
+                                feconf.CURRENT_STORY_SCHEMA_VERSION))
+
+        versioned_story_contents['schema_version'] = (
+            current_version + 1)
+
+        conversion_fn = getattr(
```

Comment:
shouldn't this be a for loop iterating over all versions one at a time from versioned_story_contents['schema_version'] to feconf.CURRENT_STORY_SCHEMA_VERSION?

Reply:
That loop is present in story_services.py and calls this entire function on each iteration of that loop.